### PR TITLE
#1290: Forbid empty changeSet id and author

### PR DESCRIPTION
**Pull request by skrivenko**
Added validation for these attributes to ValidatingVisitor.
Tests are also added for the cases when author or id or both attributes are empty.

Tags: hacktoberfest, newbiePR

### CODE REVIEW

**Comment by nvoxland** [comment]
**Pre-Review Notes**

The change looks like a nice enhancement off hand. Includes tests already. Thanks for the PR!

Questions I have:
- None

Potential risks:
- None

What could make the full review difficult:
- Nothing

**Review by MalloD12 — APPROVED** [review]
Approved.

# Review and testing results:

Overall changes and tests implemented looks good. Few things to comment:

- Update parameter from `changeSet.setValidationFailed()` call.
- Added unit tests to validate changeset `execute()` call returns the expected state based on `validationFailed` flag.
- Build and test run successfully.

### Things to be aware of:
- None

### Things to worry about:
- None

**Review by filipelautert — APPROVED** [review]

File: liquibase-core/src/test/java/liquibase/changelog/visitor/ValidatingVisitorTest.java [hunk 1]
```diff
@@ -109,6 +109,46 @@ public ValidationErrors validate(Database database) {
         assertFalse(handler.validationPassed());
     }
 
+    @Test
```
Comment (XDelphiGrl) [inline liquibase-core/src/test/java/liquibase/changelog/visitor/ValidatingVisitorTest.java:112]: Nice, complete set of unit tests. Thank you!

**Review by XDelphiGrl — APPROVED** [review]
A changeset with an empty `author` and `id` impedes the traceability provided by using Liquibase changelogs. This PR ensures that both `author` and `id` must have values. If these are not present, Liquibase will return a validation error indicating which attribute or attributes are empty.
* Complete set of unit tests added - thank you @skrivenko !
* Automated functional and test harness executions passing.
* No additional testing required.

APPROVED

**Review by suryaaki2 — APPROVED** [review]

**Comment by SandorArpa** [comment]
This change introduces a breaking change that cannot neither be fixed, nor worked around. See https://github.com/liquibase/liquibase/issues/4053

**Pull request merged**